### PR TITLE
Use Room.findPredecessor when rendering RoomCreate tiles

### PR DESCRIPTION
**Pull request by andybalaam**
Prep for [MSC3946](https://github.com/matrix-org/matrix-spec-proposals/pull/3946)

Part of https://github.com/vector-im/element-web/issues/24323

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/components/views/messages/RoomCreate.tsx [hunk 1]
```diff
@@ -37,31 +39,50 @@ interface IProps {
  * room.
  */
 export const RoomCreate: React.FC<IProps> = ({ mxEvent, timestamp }) => {
+    // Note: we ask the room for its predecessor here, instead of directly using
+    // the information inside mxEvent. This allows us the flexibility later to
+    // use a different predecessor (e.g. through MSC3946) and still display it
+    // in the timeline location of the create event.
+    const roomContext = useContext(RoomContext);
+    const predecessor = roomContext.room.findPredecessor();
```
Comment (t3chguy) [inline src/components/views/messages/RoomCreate.tsx:47]: This doesn't look like it'll react to the predecessor changing due to a new/updated state event coming in?

**Comment by andybalaam** [comment]
Tests failed with `Error: TypeError: this.client?.supportsThreads is not a function`. This seems very weird - could it be running against a wrong branch? E.g. one from https://github.com/matrix-org/matrix-react-sdk/pull/9878 ?

File: src/components/views/messages/RoomCreate.tsx (hunk 1 above)
Comment (andybalaam) [reply src/components/views/messages/RoomCreate.tsx]: Addressed in 1b4c3f6745 . Apologies for the rebase - fat-fingered it.

File: src/components/views/messages/RoomCreate.tsx [hunk 2]
```diff
@@ -37,31 +40,53 @@ interface IProps {
...
-    const prevRoom = MatrixClientPeg.get().getRoom(predecessor["room_id"]);
-    const permalinkCreator = new RoomPermalinkCreator(prevRoom, predecessor["room_id"]);
+
+    if (!predecessor) {
+        logger.warn("RoomCreate unexpectedly used in a room with no predecessor.");
+        return <div />;
+    }
+
+    const prevRoom = MatrixClientPeg.get().getRoom(predecessor.roomId);
+    const permalinkCreator = new RoomPermalinkCreator(prevRoom, predecessor.roomId);
```
Comment (t3chguy) [inline src/components/views/messages/RoomCreate.tsx:82]: Can we please not introduce strict type errors in new code

Comment (andybalaam) [reply src/components/views/messages/RoomCreate.tsx:86]: Addressed in 9b1ae17 , thank you. Remaining strict errors are in unrelated code.

File: src/components/views/messages/RoomCreate.tsx [hunk 3]
```diff
@@ -37,31 +40,57 @@ interface IProps {
...
+            "RoomCreate unexpectedly used outside of the context of the room containing this m.room.create event.",
+        );
+        return <></>;
     }
-    const prevRoom = MatrixClientPeg.get().getRoom(predecessor["room_id"]);
-    const permalinkCreator = new RoomPermalinkCreator(prevRoom, predecessor["room_id"]);
+
+    if (!predecessor) {
+        logger.warn("RoomCreate unexpectedly used in a room with no predecessor.");
+        return <div />;
```
Comment (t3chguy) [inline src/components/views/messages/RoomCreate.tsx:78]: Is that unexpected? If so can we rename it to RoomCreateWithPredecessor or something equally obvious

Comment (andybalaam) [reply src/components/views/messages/RoomCreate.tsx:78]: Yes, it is unexpected. I'd like to do the rename separately to keep this change clean.

**Review by t3chguy — APPROVED** [review]

**Pull request merged**